### DISCUSSION TITLE
Error: Invalid package reference in library; not included in Android: javax.tools. Referenced from com.squareup.kotlinpoet.FileSpec. [InvalidPackage]

### DISCUSSION THREAD

**Discussion by faelmg18** [square/kotlinpoet#1267] [General]
Hey guys.
I'm running the command .\gradlew build publishToMaveLocal, to generate a local artifact to be used in another project. But I'm having the following error in android studio

  **C:\Users\MyUser\.gradle\caches\modules-2\files-2.1\com.squareup\kotlinpoet\1.7.1\f0ff54549f263166dcc76ffde5a724a17aaa0d0\kotlinpoet-1.7.1.jar: Error: Invalid package reference in library; not included in Android: javax.tools. Referenced from com.squareup.kotlinpoet.FileSpec. [InvalidPackage]**

Can you help me please?

I'm using the following version
**com.squareup:kotlinpoet:1.7.1**

**Comment by Egorand** [comment]
This looks like Android's build toolchain complaining about a JDK package (`javax.tools`) that's not available in Dalvik? Are you using KotlinPoet inside an Android application?

**Reply by faelmg18** [reply]
I'm using an android project in android studio with a module inside the project in java

plugins {
    id("java-library")
    id("kotlin")
    id("plugin-artifact")
    id("kotlin-kapt")
}

when I do a rebuild, the jar is generated in the build/libs folder.

When I run the command **.\gradlew publishToMaveLocal**, it generates the .pom file inside the .m2 folder, but does not generate the .jar

And when I run the command
**.\gradlew build publishToMaveLocal**, gives the error I reported.

**Reply by Egorand** [reply]
Make sure your build does not package KotlinPoet into the APK, looks like it won't work due to that JDK package that's not supported on Android.
